courses/advanced-spark/chapters/concurrency.rst

```diff
@@ -105,18 +105,25 @@ Concurrency – A trivial example
 
 .. code:: ada
 
-    task type T;
+    procedure Show_Trivial_Task is
+       task type T;
```

Comment:
This cannot work with SPARK. You will get a SPARK error, as SPARK follows Ravenscar restrictions so a task has to be declared at library-level (not inside a subprogram). And in addition I think we should not show examples of procedures that do nothing just being scopes for code snippets. I changed all such examples in the basic SPARK course. BTW, the 2 pragmas described in SPARK UG (http://docs.adacore.com/spark2014-docs/html/ug/en/source/concurrency.html) should be set in a pragma configuration file (the one with currently "pragma SPARK_Mode(On);" used for all SPARK examples) or GNATprove will reject all the concurrent code.

Reply:
Yes, you're absolutely right. (I copied the wrong example from my local notebook, even though I had the correct one using packages right below. Shame on me!)